plugin/ochttp/trace.go

```diff
@@ -53,11 +56,80 @@ func (t *traceTransport) RoundTrip(req *http.Request) (*http.Response, error) {
 	name := spanNameFromURL(req.URL)
 	// TODO(jbd): Discuss whether we want to prefix
 	// outgoing requests with Sent.
-	_, span := trace.StartSpan(req.Context(), name,
+	ctx, span := trace.StartSpan(req.Context(), name,
 		trace.WithSampler(t.startOptions.Sampler),
 		trace.WithSpanKind(trace.SpanKindClient))
 
-	req = req.WithContext(trace.WithSpan(req.Context(), span))
+	tracer := &httptrace.ClientTrace{
+		GetConn: func(hostPort string) {
+			gctx, span := trace.StartSpan(ctx, "GetConn")
+			ctx = context.WithValue(gctx, "getConnSpan", span)
```

Comment:
let's create a private type to use as the key to avoid potential collisions

Reply:
Thanks, yeah this slipped my mind and I was thinking of it too.